bin/tictactoe.rb

```diff
@@ -1,40 +1,76 @@
-#defining the player class object
-class Player
-  attr_accessor :name, :id, :state
-
-  def initialize(name, id, state)
-    @name = name    #player1 or player2
-    @id =id         # 'X' or 'O'
-    @state = state  # win! or lose!
-  end
-
-  #to display the winner/loser
-  def to_s
-    "#{name} has #{state}"
-  end
+require_relative '../lib/player'
+require_relative '../bin/displayController'
+require_relative '../lib/board'
 
-end
+class TicTacToe
+  attr_reader :players, :board, :current_player, :next_player, :game_on
 
-#defining the playing board
-class Board
-  @@size = 9
-  attr_accessor :curr_value
+  def initialize(players)
+    @players = players
+    @board = Array.new(9, ' ')
+    @current_player, @next_player = players
+    @game_on = true
+  end
 
-  def initialize(curr_value)
-    @curr_value = curr_value
+  # called for player turns
+  # takes in the board values as input
+  # checks the turn value to determine what player's turn it is
+  # checks player selected position to determine if it's a valid move
+  # updates and prints the board
+  def play
+    #puts "#{@current_player.name}'s turn, pick a position between 1 - 9"
+    DisplayController.play(current_player.name)
+    pos = DisplayController.position
+    if valid_move(pos)
+      @board[pos - 1] = @current_player.id
+      switch
+      puts DisplayController.print_board(board)
```

Comment:
This `puts` should be inside the `displayController` as well.

The call made here should be just `DisplayController.print_board(board)`